sklearn/model_selection/_validation.py

```diff
@@ -389,12 +394,18 @@ def cross_val_predict(estimator, X, y=None, labels=None, cv=None, n_jobs=1,
     X, y, labels = indexable(X, y, labels)
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
+
+    # Ensure the estimator has implemented the passed decision function
+    if not hasattr(estimator, apply_func):
```

Comment:
I think I worded it wrongly. I meant after checking if `estimator` has an attribute called `apply_func`, you also need to ensure that it is a callable and not any other attribute...

```
if not(hasattr(estimator, apply_func) and callable(getattr(etimator, apply_func))):
    ...
```